In [2]:
# Import Libraries
import os
import subprocess
import stat
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from datetime import datetime
%matplotlib inline

In [3]:
#Loading the dataset
car_data = pd.read_csv('autos.csv')
car_data.head(5)

dateCrawled                            name  ... yearsOld monthsOld
0  2016-03-24 11:52:17                      Golf_3_1.6  ...       23        11
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  ...        5         7
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  ...       12         4
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  ...       15         5
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  ...        8         5

[5 rows x 27 columns]

In [6]:
# size of the dataset
car_data.shape

(313687, 27)

So, the database has 313,687 rows and 27 columns.

In [8]:
# Column names and their datatypes
car_data.dtypes

dateCrawled            object
name                   object
seller                 object
offerType              object
price                   int64
abtest                 object
vehicleType            object
yearOfRegistration      int64
gearbox                object
powerPS                 int64
model                  object
kilometer               int64
monthOfRegistration    object
fuelType               object
brand                  object
notRepairedDamage      object
dateCreated            object
postalCode              int64
lastSeen               object
yearOfCreation          int64
yearCrawled             int64
monthOfCreation        object
monthCrawled           object
NoOfDaysOnline          int64
NoOfHrsOnline           int64
yearsOld                int64
monthsOld               int64
dtype: object

**the datatype of dateCrawled, dateCreated, lastSeen can be converted from object to date as they are date fields.**

In [11]:
car_data.dateCrawled = pd.to_datetime(car_data.dateCrawled)
car_data.dateCreated = pd.to_datetime(car_data.dateCreated)
car_data.lastSeen = pd.to_datetime(car_data.lastSeen)

In [13]:
# Setting dateCRawled column as Dataframe index
car_data.set_index('dateCrawled', inplace=True)

# Sort the DataFrame by the index
car_data.sort_index(inplace=True)

In [14]:
print('Data crawling start date: ', car_data.index[0])
print('Data crawling end date: ', car_data.index[-1])
print("Total days: ", car_data.index[-1]- car_data.index[0])

Data crawling start date:  2016-03-05 14:06:22
Data crawling end date:  2016-04-07 14:36:58
Total days:  33 days 00:30:36


# **Data Cleaning**
**Check for null, missing and duplicates in the data**


In [16]:
car_data.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 313687 entries, 2016-03-05 14:06:22 to 2016-04-07 14:36:58
Data columns (total 26 columns):
 #   Column               Non-Null Count   Dtype         
---  ------               --------------   -----         
 0   name                 313687 non-null  object        
 1   seller               313687 non-null  object        
 2   offerType            313687 non-null  object        
 3   price                313687 non-null  int64         
 4   abtest               313687 non-null  object        
 5   vehicleType          313687 non-null  object        
 6   yearOfRegistration   313687 non-null  int64         
 7   gearbox              313687 non-null  object        
 8   powerPS              313687 non-null  int64         
 9   model                313687 non-null  object        
 10  kilometer            313687 non-null  int64         
 11  monthOfRegistration  313687 non-null  object        
 12  fuelType             313498 non-null  

**The only column containing null values is fuelType as the count of non null values(313498) is less than that of total rows(313687).**

In [18]:
# Checking missing values
car_data.isnull().any()

name                   False
seller                 False
offerType              False
price                  False
abtest                 False
vehicleType            False
yearOfRegistration     False
gearbox                False
powerPS                False
model                  False
kilometer              False
monthOfRegistration    False
fuelType                True
brand                  False
notRepairedDamage      False
dateCreated            False
postalCode             False
lastSeen               False
yearOfCreation         False
yearCrawled            False
monthOfCreation        False
monthCrawled           False
NoOfDaysOnline         False
NoOfHrsOnline          False
yearsOld               False
monthsOld              False
dtype: bool

In [20]:
car_data.fuelType.isnull().sum()

189

**There are 189 missing values in fuelType column Removing the rows with missing fuelType values.**

In [21]:
car_data = car_data[car_data.fuelType.notnull()]

In [24]:
car_data.duplicated().sum()

25